src/sema/types.rs

```diff
@@ -1068,7 +1068,7 @@ impl Type {
             Type::Ref(_) => false,
             Type::StorageRef(..) => false,
             Type::InternalFunction { .. } => false,
-            Type::ExternalFunction { .. } => false,
+            Type::ExternalFunction { .. } => true,
```

Comment:
On Ethereum, an external function fits into a 256 bit register, so it's not a reference type. Ideally this should be `ns.target != Target::EVM` with a comment.